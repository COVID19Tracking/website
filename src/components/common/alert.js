import React from 'react'
import classnames from 'classnames'
import alertStyle from './alert.module.scss'
import alertIcon from '~images/alert/alert.svg'

const InfoboxInner = ({ header, children }) => (
  <div>
    <p className={alertStyle.header}>{header}</p>
    <div className={alertStyle.content}>{children}</div>
  </div>
)

const Alert = ({ header, children, block = false, fullSize = false }) => (
  <div
    className={classnames(
      'alert',
      alertStyle.alert,
      fullSize && alertStyle.fullSize,
      block && alertStyle.block,
    )}
  >
    <img src={alertIcon} alt="Alert icon" />
    <InfoboxInner header={header}>{children}</InfoboxInner>
  </div>
)

export default Alert
