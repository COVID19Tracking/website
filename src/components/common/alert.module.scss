.alert {
  display: inline-flex;
  padding: spacer(8);
  padding-right: spacer(32);
  margin: spacer(16) 0;
  background-color: $color-infobox-alert-light;
  color: $color-infobox-alert-dark;
  border-radius: 4px;
  @include type-size(100);

  img {
    width: toRem(16);
    margin: spacer(4) spacer(16) spacer(8) spacer(8);
  }

  p {
    margin: 0;
  }

  .header {
    font-weight: bold;
    line-height: 1em;
  }

  .content {
    margin-top: spacer(4);
  }

  &.block {
    display: flex;
  }

  &.full-size {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding-right: spacer(8);
    text-align: center;
  }
}
