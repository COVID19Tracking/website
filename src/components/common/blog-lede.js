import React from 'react'
import Byline from './byline'
import blogLedeStyles from './blog-lede.module.scss'

export default ({ headline, author, date, lede }) => (
  <div className={blogLedeStyles.lede}>
    <h1>{headline}</h1>
    <p className="blog-lede">{lede}</p>
    <Byline author={author} date={date} />
  </div>
)
