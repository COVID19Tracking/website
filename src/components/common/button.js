import React from 'react'
import { Link } from 'gatsby'

const ButtonLink = ({ big, to, children }) => (
  <Link to={to} className={`button${big ? ' button-big' : ''}`}>
    {children}
  </Link>
)

const ButtonAnchor = ({ big, href, children }) => (
  <a href={href} className={`button${big ? ' button-big' : ''}`}>
    {children}
  </a>
)

const Button = ({ big, children }) => (
  <button className={`button${big ? ' button-big' : ''}`} type="button">
    {children}
  </button>
)

export { ButtonAnchor, ButtonLink, Button }
