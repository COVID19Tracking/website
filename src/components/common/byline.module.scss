.byline {
  margin: 0.3rem 0;
  font-size: 1rem;
  a {
    font-weight: 700;
  }
  .date {
    display: inline-block;
    margin-left: 1rem;
    color: #808080;
  }
  font-size: 0.95rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
