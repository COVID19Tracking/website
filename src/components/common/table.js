import React from 'react'
import DetailText from './detail-text'

const Table = ({ children, className, tableLabel }) => (
  <div>
    <table className={className}>{children}</table>
    {tableLabel && <DetailText>{tableLabel}</DetailText>}
  </div>
)

export default Table
