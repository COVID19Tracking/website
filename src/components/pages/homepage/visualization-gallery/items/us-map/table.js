import React from 'react'
import { Link } from 'gatsby'
import slugify from 'slugify'

const Table = ({ states, metric, us }) => (
  <div className="a11y-only">
    <Link to={`#skip-state-list-${slugify(metric.title)}`}>
      Skip the list of states
    </Link>
    <table>
      <thead>
        <tr>
          <th>State</th>
          <th>{metric.title}</th>
          <th>Total test results</th>
          <th>Total cases</th>
          <th>Currently hospitalized</th>
          <th>Deaths</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>National total</td>
          <td>{Math.round(us.value)}</td>
          <td>{us.current.totalTestResults}</td>
          <td>{us.current.positive}</td>
          <td>N/A</td>
          <td>{us.current.death}</td>
        </tr>
        {states.map(state => (
          <tr>
            <td>
              <Link to={`/data/state/${state.childSlug.slug}`}>
                {state.name}
              </Link>
            </td>
            <td>{Math.round(state.value)}</td>
            <td>{state.current.totalTestResults}</td>
            <td>{state.current.positive}</td>
            <td>{state.current.hospitalizedCurrently}</td>
            <td>{state.current.death}</td>
          </tr>
        ))}
      </tbody>
    </table>
    <div id={`skip-state-list-${slugify(metric.title)}`} />
  </div>
)

export default Table
