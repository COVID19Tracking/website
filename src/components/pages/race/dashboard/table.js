import React from 'react'
import { Link } from 'gatsby'
import Tooltip from '~components/common/tooltip'
import { Table, Th, Td } from '~components/common/table'
import Percent from './percent'
import { Notes, DisparitySymbol, TooltipContent } from './table-symbols'
import tooltipDisparityIcon from '~images/race-dashboard/tooltip-disparity-icon.svg'
import alertBang from '~images/race-dashboard/alert-bang-orange.svg'
import stateTableStyle from './table.module.scss'

const StateTable = ({ children }) => (
  <Table className={stateTableStyle.table}>{children}</Table>
)

const GroupRowHeader = ({ children }) => (
  <Th additionalClass={stateTableStyle.group} scope="row">
    {children}
  </Th>
)

const StateTableDataHeader = ({ noData, children }) => {
  const classes = [stateTableStyle.percent]
  if (noData) {
    classes.push(stateTableStyle.missingData)
  }
  return (
    <Th additionalClass={classes}>
      {noData && <img src={alertBang} alt="Alert icon" />}
      <span className={stateTableStyle.headerText}>
        {children}
        {!noData && <div className={stateTableStyle.symbolSpacer} />}
      </span>
    </Th>
  )
}

const StateTableHeader = ({
  groupTitle,
  noDeaths,
  noPositives,
  noTests,
  noHospitalizations,
}) => (
  <thead>
    <tr>
      <Th additionalClass={stateTableStyle.group}>{groupTitle}</Th>
      <Th additionalClass={stateTableStyle.percent} isFirst>
        Percentage of
        <br />
        population
      </Th>
      <StateTableDataHeader noData={noPositives}>
        Percentage of
        <br />
        cases
      </StateTableDataHeader>
      <StateTableDataHeader noData={noDeaths}>
        Percentage of
        <br />
        deaths
      </StateTableDataHeader>
      <StateTableDataHeader noData={noTests}>
        Percentage of
        <br />
        tests
      </StateTableDataHeader>
      <StateTableDataHeader noData={noHospitalizations}>
        Percentage of
        <br />
        hospitalizations
      </StateTableDataHeader>
    </tr>
  </thead>
)

const StateCellPercent = ({ number, disparity, note, noteIndex }) => {
  const value = <Percent number={number} highlight={disparity || note} />
  if (disparity) {
    return (
      <Tooltip
        label={
          <>
            <img src={tooltipDisparityIcon} alt="" /> Racial/ethnic disparity
            likely.
          </>
        }
      >
        <span>{value}</span>
      </Tooltip>
    )
  }
  if (note) {
    return (
      <Tooltip label={<TooltipContent title={note} index={noteIndex} />}>
        <span>{value}</span>
      </Tooltip>
    )
  }
  return value
}

const StateTableDataCell = ({
  index,
  rowCount,
  state,
  stateAbbr,
  type,
  errorType,
  noData, // if true, this data is not provided by this state
  cellData,
}) => {
  if (noData) {
    if (index === 0) {
      return (
        <Td rowspan={rowCount} additionalClass={stateTableStyle.missingData}>
          <span>
            {state} does not report {type.toLowerCase()} data for {errorType}.{' '}
            <Link to="/race/get-better-data">Help us get better data.</Link>
          </span>
        </Td>
      )
    }
    return <></>
  }
  return (
    <Td>
      <>
        <StateCellPercent
          // todo remove || for number
          number={cellData.value || undefined}
          disparity={cellData.disparity}
          note={cellData.note.value}
          noteIndex={cellData.note.index + 1}
        />

        <div className={stateTableStyle.symbolSpacer}>
          <Notes
            index={cellData.note.index + 1}
            title={cellData.note.value}
            linkTo={`notes-${stateAbbr.toLowerCase()}`}
          />
          {cellData.disparity && (
            <DisparitySymbol linkTo={`notes-${stateAbbr.toLowerCase()}`} />
          )}
        </div>
      </>
    </Td>
  )
}

const StateTableBody = ({
  state,
  stateAbbr,
  rows,
  type,
  noPositives = false,
  noDeaths = false,
  noTests = false,
  noHospitalizations = false,
}) => (
  <tbody>
    {rows.map((row, index) => (
      <tr>
        <GroupRowHeader>{row.group}</GroupRowHeader>
        <Td isFirst>
          <Percent number={row.population} />
        </Td>
        <StateTableDataCell
          index={index}
          rowCount={rows.length}
          type={type}
          errorType="cases"
          state={state}
          stateAbbr={stateAbbr}
          noData={noPositives}
          cellData={row.positive}
        />
        <StateTableDataCell
          index={index}
          rowCount={rows.length}
          type={type}
          errorType="deaths"
          state={state}
          stateAbbr={stateAbbr}
          noData={noDeaths}
          cellData={row.death}
        />
        {row.test && (
          <StateTableDataCell
            // todo remove conditional here
            index={index}
            rowCount={rows.length}
            type={type}
            errorType="tests"
            state={state}
            stateAbbr={stateAbbr}
            noData={noTests}
            cellData={row.test}
          />
        )}
        {row.hospitalization && (
          // todo remove conditional here
          <StateTableDataCell
            index={index}
            rowCount={rows.length}
            type={type}
            errorType="hospitalizations"
            state={state}
            stateAbbr={stateAbbr}
            noData={noHospitalizations}
            cellData={row.hospitalization}
          />
        )}
      </tr>
    ))}
  </tbody>
)

export { StateTable, GroupRowHeader, StateTableHeader, Td, StateTableBody }
