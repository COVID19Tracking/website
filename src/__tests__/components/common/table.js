import React from 'react'
import renderer from 'react-test-renderer'
import { Table } from '../../../components/common/table'

describe('Components : Common: Table', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Table>
          <thead>
            <tr>
              <th>Column A</th>
              <th>Column B</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Item A</td>
              <td>Item B</td>
            </tr>
            <tr>
              <td>Item A</td>
              <td>Item B</td>
            </tr>
            <tr>
              <td>Item A</td>
              <td>Item B</td>
            </tr>
          </tbody>
        </Table>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
