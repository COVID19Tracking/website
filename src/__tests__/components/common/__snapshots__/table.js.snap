// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Common: Table renders correctly 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Column A
        </th>
        <th>
          Column B
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Item A
        </td>
        <td>
          Item B
        </td>
      </tr>
      <tr>
        <td>
          Item A
        </td>
        <td>
          Item B
        </td>
      </tr>
      <tr>
        <td>
          Item A
        </td>
        <td>
          Item B
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
